### ISSUE TITLE
Switch DB implementation to rocksdb

### ISSUE THREAD

**Issue by chfast** [ethereum/aleth#4037]
I would like to try rocksdb instead of leveldb. The most important advantage is actually Windows support. They promise to have the same API and we had rocksdb support before (as an option).

Before starting we should add some performance test to it, like block imports, to compare later on.

Also, do #4036 first to have nice build experience from the beginning.

**Comment by gumb0** [comment]
@halfalicious See also my work in https://github.com/ethereum/aleth/pull/4844

**Comment by halfalicious** [comment]
No longer needed as we've switched to pulling in LevelDB via Hunter so we're using the same implementation on all platforms.
